sentry_sdk/integrations/httpx.py

```diff
@@ -52,13 +53,20 @@ def send(self, request, **kwargs):
             span.set_data("http.query", parsed_url.query)
             span.set_data("http.fragment", parsed_url.fragment)
 
-            for key, value in hub.iter_trace_propagation_headers():
-                logger.debug(
-                    "[Tracing] Adding `{key}` header {value} to outgoing request to {url}.".format(
-                        key=key, value=value, url=request.url
+            if should_propagate_trace(
+                str(request.url),
+                hub.client.options["trace_propagation_targets"]
```

Comment:
I have now implemented a suggestion from markus below, which makes the code way easier.
I rather not do the "fetching the config" in function, so the function does just does one thing, check if the trace should be for the given url and trace_propagation_targets.

Reply:
I was thinking a bit. How about we do a `hub.should_propagate_trace(url)` method? Is the hub the right "thing" to ask if the the trace to a given url should be propagated?

Reply:
Maybe you @untitaker also have an opinion about my idea above?

Reply:
I don't have a strong opinion on this at all, I think there's no strong pattern to follow here. We do however have a few freestanding functions lying around the entire codebase that just take a hub as their first argument. That might be a bit better because then you can at least define `should_propagate_trace` in some helper module rather than the ever-growing `hub.py`